vagrant/roles/pxe-server/tasks/main.yml

```diff
@@ -0,0 +1,143 @@
+---
+- name: install required RPMs
+  package:
+    name: "{{item}}"
+    state: present
+  become: yes
+  with_items: "{{required_rpms}}"
+
+- name: create configuration file for eth1
+  template:
+    src: ifcfg-eth1.j2
+    dest: /etc/sysconfig/network-scripts/ifcfg-eth1
+  become: yes
+
+- name: restart network
+  service:
+    name: network
+    state: restarted
+  become: true
+
+- name: modify configuration file of dhcp server
+  template:
+    src: dhcpd.conf.j2
+    dest: /etc/dhcp/dhcpd.conf
+  become: yes
+
+- name: enable dhcpd service
+  service:
+    name: dhcpd
+    enabled: yes
+  become: true
+
+- name: start dhcpd service
+  service:
+    name: dhcpd
+    state: started
+  become: true
+
+- name: enable tftp.socket
+  service:
+    name: tftp.socket
+    enabled: yes
+  become: true
+
+- name: start tftp.socket
+  service:
+    name: tftp.socket
+    state: started
+  become: true
+
+- name: enable vsftpd service
+  service:
+    name: vsftpd
+    enabled: yes
+  become: true
+
+- name: start vsftpd service
+  service:
+    name: vsftpd
+    state: started
+  become: true
+
+- name: create directory /vagrant_data
+  file:
+    path: /vagrant_data
+    state: directory
+    owner: vagrant
+    group: vagrant
+    mode: 0755
+  become: true
+
+- name: check if installation iso image is already downloaded
+  stat:
+    path: "/vagrant_data/{{iso_img_file}}"
+  register: iso_img_stat_result
```

Comment:
This is unnecessary as get_url will not re-download the file :smile:

Reply:
Sure :smiley: